app/models/katello/erratum.rb

```diff
@@ -144,6 +144,30 @@ def module_stream_objects
       return streams.flatten.uniq
     end
 
+    def self.import_for_repository(repository)
+      pulp_ids = []
+      service_class = SmartProxy.pulp_master!.content_service(content_type)
+      fetch_only_ids = !repository.content_view.default? &&
+          !repository.repository_type.unique_content_per_repo
+      service_class.pulp_units_batch_for_repo(repository, fetch_identifiers: fetch_only_ids).each do |units|
+        units.each do |unit|
+          unit = unit.with_indifferent_access
+          errata_id = unit[service_class.unit_identifier]
+          unless fetch_only_ids
+            model = Katello::Util::Support.active_record_retry do
+              self.where(:errata_id => errata_id).first_or_create(:pulp_id => errata_id)
+            end
+            service = service_class.new(model.errata_id)
+            service.backend_data = unit
+            model.repository_id = repository.id unless many_repository_associations
+            service.update_model(model)
+          end
+          pulp_ids << errata_id
+        end
+      end
+      sync_repository_associations(repository, :errata_ids => pulp_ids) if self.many_repository_associations
```

Comment:
shouldn't that be :pulp_ids => pulp_ids? ?

Reply:
or else, shouldn't sync_repository_associations be enhanced to handle :errata_ids ?

Reply:
i think it would make more sense to just pull the pulp_id from the errata (looked up via the errata_id) and use sync_repository_associations as is?

Reply:
In which case you might be able to distill the differences into just a method that does something like, unless i'm missing other differences:
```
               self.where(:errata_id => errata_id).first_or_create(:pulp_id => errata_id)
```

Reply:
hmm..looks like this PR is missing a commit..I am trying to add errata_id support to sync_repository_associations.

Reply:
Actually..I think the changes to the sync_repository_associations are in the pulp_database_unit file above..

Reply:
> In which case you might be able to distill the differences into just a method that does something like, unless i'm missing other differences:
> 
> ```
>                self.where(:errata_id => errata_id).first_or_create(:pulp_id => errata_id)
> ```

I'll test this out..

Reply:
oh hah, sorry i missed it!  but yes, i'd like to see if there's a way to not have to duplicate this method completely

Reply:
I think as long as the pulp2_3 migrated errata has pulp_id as errata_id instead of pulp_href, setting identifier to id for errata should work fine thanks to this: https://github.com/Katello/katello/blob/b5e7d05c134a864e8f1971b803fd2fadc8648bb0/app/models/katello/concerns/pulp_database_unit.rb#L112

Added a separate commit to try that.